packages/components/src/RadioGroup/RadioGroup.tsx

```diff
@@ -30,22 +35,34 @@ interface RadioGroupProps {
 export function RadioGroup({
   children,
   value,
+  label,
   onChange,
   name = uuid.v1(),
 }: RadioGroupProps) {
+  const labelId = `${name}_label`;
+
   return (
-    <div role="radiogroup" className={styles.radioGroup}>
-      {React.Children.map(children, option => (
-        <InternalRadioOption
-          checked={value === option.props.value}
-          name={name}
-          onChange={handleChange}
-          {...option.props}
-        >
-          {option.props.children}
-        </InternalRadioOption>
-      ))}
-    </div>
+    <>
+      <span id={labelId} className={styles.groupLabel}>
+        {label}
+      </span>
```

Comment:
@chris-at-jobber i know you had created the issue for this PR, I'm curious if the fieldset and legend might be better suited for this use case as it appears more semantically correct?

Reply:
@joelwarrington-jobber I've seen the same approach taken for other component libraries as well. I'd be down for swapping out the markup. Let me know what y'all decide.